server/backend/database/memory/database.go

```diff
@@ -160,16 +175,48 @@ func (d *DB) ListProjectInfos(ctx context.Context) ([]*database.ProjectInfo, err
 }
 
 // UpdateProjectInfo updates the given project.
-func (d *DB) UpdateProjectInfo(ctx context.Context, info *database.ProjectInfo) error {
+func (d *DB) UpdateProjectInfo(
+	ctx context.Context,
+	id types.ID,
+	fields *types.UpdatableProjectFields,
+) (*database.ProjectInfo, error) {
 	txn := d.db.Txn(true)
 	defer txn.Abort()
 
+	// field.Name unique test
+	exist, err := txn.First(tblProjects, "name", *fields.Name)
+	if err != nil {
+		return nil, err
+	}
+	if exist != nil {
+		return nil, fmt.Errorf("%s: %w", *fields.Name, database.ErrProjectNameAlreadyExists)
+	}
```

Comment:
`fields.Name` can be `nil`.
```suggestion
	if (fields.Name != nil) {
		exist, err := txn.First(tblProjects, "name", *fields.Name)
		if err != nil {
			return nil, err
		}
		if exist != nil {
			return nil, fmt.Errorf("%s: %w", *fields.Name, database.ErrProjectNameAlreadyExists)
		}
	}
```